Problem2.java

```diff
@@ -0,0 +1,43 @@
+// Time Complexity :
+//		modBinSearch() - O(log(n))
+//      
+// Space Complexity :
+//		modBinSearch() - O(1)
+//
+// Did this code successfully run on Leetcode : N/A
+// Any problem you faced while coding this : No
+
+class Problem2
+{
+	public static void main(String[] args)
+	{
+		int[] num = new int[]{1,3,5,7,8,9,10,11,13,14,15,16,17,18};
+		int target = 14;
+
+		System.out.println("Modified Binary Search(14) -> " + modBinSearch(num, target));
+	}
+
+	private static int modBinSearch(int[] nums, int target)
+	{
+		int l=0, h=1;
+
+		while(h < nums.length && target >= nums[h])
+		{
+			l = h;
+			h = 2*h;
```

Comment:
how will you handle the case when h is less than nums.length but when this condition will be satisfied we will have h as 2*h but in some cases 2*h will be out of range